src/mlpack/methods/preprocess/load_save_image_main.cpp

```diff
@@ -0,0 +1,139 @@
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core.hpp>
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+using namespace mlpack::data;
+
+
+PROGRAM_INFO("Load Save Image",
+    // Short description.
+    "A utility to load and save image dataset. This utility will allow you to "
+    "load and save a single image or an list of images.",
+    // Long description.
+    "This utility takes a image or an array of images and loads them to a"
+    " matrix. You can specify the height " + PRINT_PARAM_STRING("height") +
+    " width " + PRINT_PARAM_STRING("width") + " and channel " +
+    PRINT_PARAM_STRING("channel") + " of the images that needs to be loaded. "
+    "\nThere are other options too, that can be specified such as " +
+    PRINT_PARAM_STRING("quality") + " and " + PRINT_PARAM_STRING("transpose")
+    + ".\n\n" +
+    "You can also provide a dataset and save them as images using " +
+    PRINT_PARAM_STRING("dataset") + " and " + PRINT_PARAM_STRING("save") +
+    " as an parameter. An example to load an image : "  + "\n\n" +
+    PRINT_CALL("load_save_image", "input", "X", "height", 256, "width", 256,
+        "channel", 3, "output", "Y") + "\n\n" +
+    " An example to save an image is :" + "\n\n" +
+    PRINT_CALL("load_save_image", "input", "X", "height", 256, "width", 256,
+        "channel", 3, "dataset", "Y", "save", true) + "\n\n" +
+    " An example to load an image and also flipping it while loading is :"
+    + "\n\n" +
+    PRINT_CALL("load_save_image", "input", "X", "height", 256, "width", 256,
+        "channel", 3, "output", "Y", "transpose", true),
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+// DEFINE PARAM
+PARAM_VECTOR_IN_REQ(string, "input", "Image filenames which has to "
+    "be loaded/saved", "i");
+
+PARAM_INT_IN("width", "Width of the image", "W", 256);
+PARAM_INT_IN("channel", "Number of channel", "C",  3);
+
+PARAM_MATRIX_OUT("output", "Matrix to save images data to.", "o");
+
+PARAM_INT_IN("quality", "Compression of the image if saved as jpg (0-100).",
+    "q", 90);
+
+PARAM_FLAG("transpose", "Loaded dataset to be transposed", "t");
+
+PARAM_INT_IN("height", "Height of the images", "H", 256);
+PARAM_FLAG("save", "Save a dataset as images", "s");
+PARAM_MATRIX_IN("dataset", "Input matrix to save as images.", "I");
+
+// Loading/saving of a Image Info model.
+PARAM_MODEL_IN(ImageInfo, "input_model", "Input Image Info model.", "m");
+PARAM_MODEL_OUT(ImageInfo, "output_model", "Output Image Info model.", "M");
```

Comment:
I guess the functionality is ok to be there, no harm to us either, May be someone who don't have a good memory ( I don't :octocat:, and that's the sole reason I came up with because I tend to forget it while testing ) can use it :)